nats-compiler-plugin/src/main/java/io/ballerina/stdlib/nats/plugin/PluginConstants.java

```diff
@@ -0,0 +1,105 @@
+    public static final String ERROR_OR_NIL = ERROR + NIL;
+    public static final String NIL_OR_ERROR = "()|" + ERROR;
+    public static final String NATS_ERROR_OR_NIL = NATS_ERROR + NIL;
+    public static final String NIL_OR_NATS_ERROR = "()|" + NATS_ERROR;
+    static final String[] ANY_DATA_RETURN_VALUES = {ERROR, NATS_ERROR, ERROR_OR_NIL, NATS_ERROR_OR_NIL,
+            NIL_OR_ERROR, NIL_OR_NATS_ERROR, "string", "int", "float", "decimal", "boolean", "xml", "anydata",
+            "string|error", "int|error", "float|error", "decimal|error", "boolean|error", "xml|error", "anydata|error",
+            "error|string", "error|int", "float|error", "error|decimal", "error|boolean", "error|boolean",
+            "error|anydata", "string?", "anydata?", "int?", "float?", "decimal?", "xml?", "boolean?"};
+
+    /**
+     * Compilation errors.
+     */
+    enum CompilationErrors {
+        ON_MESSAGE_OR_ON_REQUEST("Only one of either onMessage or onRequest is allowed.", "NATS_101"),
+        NO_ON_MESSAGE_OR_ON_REQUEST("Service must have either remote function onMessage or onRequest.",
+                "NATS_102"),
+        INVALID_REMOTE_FUNCTION("Invalid remote function.", "NATS_103"),
+        FUNCTION_SHOULD_BE_REMOTE("Function must have the remote qualifier.", "NATS_104"),
+        MUST_HAVE_MESSAGE("Must have the function parameter nats:Message.", "NATS_105"),
+        MUST_HAVE_MESSAGE_AND_ERROR("Must have the function parameters nats:Message and nats:Error.",
+                "NATS_106"),
+        INVALID_FUNCTION_PARAM("Invalid function parameter.", "NATS_107"),
+        INVALID_FUNCTION_PARAM_MESSAGE("Invalid function parameter. Only nats:Message is allowed.",
+                "NATS_108"),
+        INVALID_FUNCTION_PARAM_ERROR("Invalid function parameter. Only nats:Error is allowed.",
+                "NATS_109"),
+        ONLY_PARAMS_ALLOWED("Invalid function parameter count. Only nats:Message is allowed.",
+                "NATS_110"),
+        ONLY_PARAMS_ALLOWED_ON_ERROR("Invalid function parameter count. Only nats:Message and nats:Error are allowed.",
+                "NATS_111"),
+        INVALID_RETURN_TYPE_ERROR_OR_NIL("Invalid return type. Only error? or nats:Error? is allowed.",
+                "NATS_112"),
+        INVALID_RETURN_TYPE_ANY_DATA("Invalid return type. Only anydata or error is allowed.",
+                "NATS_113"),
+        INVALID_MULTIPLE_LISTENERS("Multiple listener attachments. Only one nats:Listener is allowed.",
+                "NATS_114"),
+        INVALID_ANNOTATION_NUMBER("Only one service config annotation is allowed.",
+                "NATS_115"),
+        NO_ANNOTATION("No @nats:ServiceConfig{} annotation is found.",
+                "NATS_116"),
+        INVALID_ANNOTATION("Invalid service config annotation. Only @nats:ServiceConfig{} is allowed.",
+                "NATS_117"),
+        INVALID_SERVICE_NAME("Invalid service name. Only string literals are allowed.",
```

Comment:
What's "service name" here? The term is not exactly accurate since now we don't have a notion of a service name